bundles/test/org.eclipse.smarthome.test/src/main/java/org/eclipse/smarthome/test/java/JavaOSGiTest.java

```diff
@@ -245,19 +253,15 @@ private BundleContext initBundleContext() {
      * Unregister an OSGi service by the given object, that was registered before.
      *
      * @param interfaceName the interface name of the service
-     * @return the service registration that was unregistered or null if no service could be found
+     * @return the first service registration that was unregistered or null if no service could be found
      */
     protected ServiceRegistration<?> unregisterService(final String interfaceName) {
-        final ServiceRegistration<?> reg = registeredServices.remove(interfaceName);
-        if (reg != null) {
-            reg.unregister();
-            Iterator<ServiceRegistration<?>> regs = registeredServices.values().iterator();
-            while (regs.hasNext()) {
-                final ServiceRegistration<?> otherReg = regs.next();
-                if (otherReg == reg) {
-                    regs.remove();
-                }
-            }
+        ServiceRegistration<?> reg = null;
+        List<ServiceRegistration<?>> regList = registeredServices.remove(interfaceName);
+        if (regList != null) {
+            reg = regList.get(0);
+            registeredServices.get(interfaceName).forEach(r -> r.unregister());
```

Comment:
You just removed the map entry for `interfaceName` so this call will result in NPE.

Reply:
ah, damn. Copy/Paste error.